### DISCUSSION TITLE
Task01 Problem 2

### DISCUSSION THREAD

**Discussion by ryankong123** [chulminy/CIVE497-CIVE700#115] [General]
Hi, I'm having some difficulty understanding the Colour Images code in Task 01 Problem 2. I don't really understand the following portions of the code.

imSize = [size(imgC,1) size(imgC,2)];
imgCR = imgC(:,:,1);
imgCG = imgC(:,:,2);
imgCB = imgC(:,:,3);

and

% three dimension matrix
imgCRImg = cat(3,imgCR, zeros(imSize, 'uint8'), zeros(imSize, 'uint8')); % how to use cat
imgCGImg = cat(3, zeros(imSize, 'uint8'), imgCG, zeros(imSize, 'uint8'));
imgCBImg = cat(3, zeros(imSize, 'uint8'), zeros(imSize, 'uint8'), imgCB);
subplot(131); imshow(imgCRImg); 
subplot(132); imshow(imgCGImg); 
subplot(133); imshow(imgCBImg); 
figure; imshow(imgCRImg+imgCGImg+imgCBImg); 

I understand we are showing the red, green, and blue channels of the image separately and then showing all of them combined, but I don't really understand how the code is doing that. Any help is much appreciated!

**Comment by chulminy** [comment]
Colors can be represented by a combination of RGB values. 

Each R, G, and B values take a value between 0 and 255 because the color values are quantized with 8 bit. 

Thus, color images have a 3D matrix in an uint8 type. 

The dimension of the image is image height x image width x 3 

imgCR = imgC(:,:,1) means you read red values for all pixels. 
imgCG = imgC(:,:,2) means you read blue values for all pixels. 
imgCB = imgC(:,:,3) means you read green values for all pixels. 

Here, imgCR, imgCG, and imgCB are NOT color images. The color images must have 3 depth. 

So, to represent a red color image, you need to set the pixels of blue, and green depths as zeros. 

imgCRImg = cat(3,imgCR, zeros(imSize, 'uint8'), zeros(imSize, 'uint8')); % how to use cat

cat function is stacking the same size matrix in a depth direction. 

Since green and blue matrix in depth 2 and 3 becomes 0, imgCRImg represent a pure red image. 

Does it make sense?

**Reply by ryankong123** [reply]
Yes, thanks so much for clarifying!
